ap/ap.go

```diff
@@ -72,14 +72,24 @@ func (ap *Accesspoint) init() (err error) {
 	}
 
 	// open connection to accesspoint
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
-	defer cancel()
-	ap.conn, err = proxy.Dial(ctx, "tcp", ap.addr())
-	if err != nil {
-		return fmt.Errorf("failed dialing accesspoint: %w", err)
+	attempts := 0
+	for {
+		attempts++
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+		defer cancel()
```

Comment:
This defer will trigger only when returning from the function therefore we might have up to 6 contexts running simultaneosly. This shouldn't be an issue, just saying.